src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -118,6 +118,10 @@ def search_outdated(
     def mark_outdated_nodes(cls) -> None:
         outdated = cls.search_outdated()
         for node in outdated:
+            if node.reimage_requested or node.delete_requested:
```

Comment:
Instead of doing this when iterating, would it be better to have the `search_outdated()` method only return such nodes, or if it is used elsewhere, add a new method for querying just those nodes?